src/Raven.Server/Rachis/RachisConsensus.cs

```diff
@@ -191,26 +191,48 @@ public void Record(string message)
 
     public class RachisDebug
     {
-        public readonly ConcurrentDictionary<string, ConcurrentQueue<RachisTimings>> TimingTracking = new ConcurrentDictionary<string, ConcurrentQueue<RachisTimings>>();
+        public readonly ConcurrentDictionary<string, RachisTimingsHolder> TimingTracking = new ConcurrentDictionary<string, RachisTimingsHolder>();
         public readonly ConcurrentQueue<string> StateChangeTracking = new ConcurrentQueue<string>();
 
+        public class RachisTimingsHolder
+        {
+            public ConcurrentQueue<RachisTimings> TimingTracking;
+            public DateTime Since = DateTime.UtcNow;
+        }
+
         public bool IsInterVersionTest;
 
         public RachisLogRecorder GetNewRecorder(string name)
         {
-            var queue = new ConcurrentQueue<RachisTimings>();
-            if (TimingTracking.TryAdd(name, queue) == false)
+            var holder = new RachisTimingsHolder
+            {
+                TimingTracking = new ConcurrentQueue<RachisTimings>()
+            };
+
+            if (TimingTracking.TryAdd(name, holder) == false)
             {
                 throw new ArgumentException($"Recorder with the name '{name}' already exists");
             }
-            return new RachisLogRecorder(queue);
+            return new RachisLogRecorder(holder.TimingTracking);
+        }
+
+        public void RemoveRecorderOlderThan(DateTime after)
+        {
+            var toRemove = TimingTracking.Where(x => x.Value.Since <= after).Select(x => x.Key);
```

Comment:
Better yet, move the linq statement to the `foreach`

Reply:
wasn't sure it is safe to do. 
done.